internal/runners/branch/add.go

```diff
@@ -0,0 +1,59 @@
+package branch
+
+import (
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/project"
+)
+
+type Add struct {
+	out     output.Outputer
+	project *project.Project
+}
+
+type AddParams struct {
+	Label string
+}
+
+func NewAdd(prime primeable) *Add {
+	return &Add{
+		out:     prime.Output(),
+		project: prime.Project(),
+	}
+}
+
+func (a *Add) Run(params AddParams) error {
+	logging.Debug("ExecuteAdd")
+
+	project, err := model.FetchProjectByName(a.project.Owner(), a.project.Name())
+	if err != nil {
+		return locale.WrapError(err, "branch_list_proejct_err", "Could not get project details for project: {{.V0}}", a.project.Namespace().String())
+	}
+
+	branchID, err := model.AddBranch(project.ProjectID, params.Label)
+	if err != nil {
+		return locale.WrapError(err, "err_add_branch", "Could not add branch")
+	}
+
+	localBranch := a.project.BranchName()
+	branch, err := model.BranchForProjectByName(project, localBranch)
+	if err != nil {
+		return locale.WrapError(err, "err_add_branch_no_default", "Could not retrieve branch information for: {{.V0}}", localBranch)
+	}
+
+	err = model.UpdateBranchTracking(*branchID, a.project.CommitUUID(), branch.BranchID, model.TrackingIgnore)
```

Comment:
`*branchID` wasn't nil checked.

Reply:
I opted to update the return type from `AddBranch`